src/app/core/utils/math/BezierCurve.ts

```diff
@@ -3,16 +3,58 @@ import {Transform} from "./Transform";
 import {Clamp} from "./MathUtils";
 import {serializable} from "serialeazy";
 
-@serializable("BezierCurve")
+/**
+ * Cubic Bezier Curve class
+ * 
+ * Consists of a start point, an end point,
+ * 2 control points, and the bounding
+ * box for the curve
+ * 
+ * Link to an interactive cubic bezier curve with formulas:
+ * https://www.desmos.com/calculator/rptlhv5rx8
+ */
+ @serializable("BezierCurve")
 export class BezierCurve {
+
+    /**
+     * the x, y coordinates of the start point
+     */
     private p1: Vector;
+
+    /**
+     * the x, y coordinates of the end point
+     */
     private p2: Vector;
+
+    /**
+     * the x, y coordinates of the first control point
+     */
     private c1: Vector;
+
+    /**
+     * the x, y coordinates of the second control point
+     */
     private c2: Vector;
 
+    /**
+     * the Bounding Box that encases the entire curve
+     */
     private boundingBox: Transform;
+
+    /**
+     * whether the curve's data has been updated
+     */
     private dirty: boolean;
 
+    /**
+     * Initializes bezier curve with the given start point, end point, and intermediate points.
+     * If no point(s) are given, then it will be initialized with a blank Vector.
+     * Bounding Box holds meaningless values by default
+     * @param p1 Initializes start point with given coordinates
```

Comment:
Looking much better now! Just make sure to leave a new line between the end of the description and the first tag, so this section would look like
```
     * Bounding Box holds meaningless values by default
     * 
     * @param p1 Initializes start point with given coordinates
```